fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/ConnectingActivity.java

```diff
@@ -126,11 +120,17 @@ private void handleConnecting(boolean isDestinationHome) {
     if (isCreation) {
       String laoName = getIntent().getExtras().getString(Constants.LAO_NAME);
       List<String> witnessesList = getIntent().getStringArrayListExtra(Constants.WITNESSES);
+      boolean isWitnessingEnabled =
+          getIntent().getExtras().getBoolean(Constants.WITNESSING_FLAG_EXTRA);
       List<PublicKey> witnesses =
-          witnessesList.stream().map(PublicKey::new).collect(Collectors.toList());
+          isWitnessingEnabled
+              ? witnessesList.stream().map(PublicKey::new).collect(Collectors.toList())
+              : Collections.emptyList();
 
       // Add the organizer to the list of witnesses
-      witnesses.add(keyManager.getMainPublicKey());
+      if (isWitnessingEnabled) {
+        witnesses.add(keyManager.getMainPublicKey());
+      }
```

Comment:
This could be simplified to be inside the same `if(isWitnessingEnabled) {..}` statement to make it more clear.